lib/pure/random.nim

```diff
@@ -204,7 +204,7 @@ proc skipRandomNumbers*(s: var Rand) =
   s.a0 = s0
   s.a1 = s1
 
-proc rand*(r: var Rand; max: Natural): int {.benign.} =
+proc rand*[T: uint | uint64](r: var Rand; max: T): T =
```

Comment:
When we need `gcsafe`? Should we mark all as `{.benign.}` or at least `{.gcsafe.}`?

Reply:
benign seems to be used very randomly (or at least I'm not understanding when it gets used vs not); I've removed it because we can rely on effect system to infer those.

Reply:
Is the `{.benign.}` pragma documented? What is your function?

Reply:
> Is the {.benign.} pragma documented?

https://github.com/nim-lang/Nim/issues/10123

Reply:
If you need uint64 to fix the bug internally this doesn't have to imply an API addition.

Reply:
fixed